build(release): add reno tools to build image and makefile (#2604)
kanisterio/kanister: CODE_REVIEW.md

# Code review requiements

> This document describe responsibilities of code reviewers when reviewing PRs.

Reviewer: ```suggestion
This document describes responsibilities of code reviewers when reviewing PRs.
```

Pull request process is described in [contributing guide](./CONTRIBUTING.md#submitting-pull-requests)

## Base checklist

- All automated test steps pass (e.g. tests, lints, build)
- PR title follows [commit conventions](CONTRIBUTING.md#commit-conventions)
    - If PR format is different, reviewer should change it to follow the conventions
- PR has a description with reasoning and change overview

>     - If description is missing but clear for reviewer, reviewer may add a comment or edit description

Reviewer: ```suggestion
    - If description is missing but clear for reviewer, reviewer may request the author to add the description or edit description by themselves
```

- New feature or fix has tests proving it works
    - Reviewer should request changes from contributor to add tests

> - If PR changes documented behaviour, there should be update to `/docs` or inline comments

Reviewer: It took me a second to realise what exactly we meant by `If PR changes documented behaviour`. Should we try to simplify it.
`If the PR changes a behaviour that is documented in our public docs`?
This should be applicable for something that is being  added (not updated)  and the documentation needs to be added for that, I think we should specify this as well.

    - Reviewer should request changes from contributor to update docs
- If PR introduces breaking changes, fixes a bug or adds a new feature, there should be a [release note](#release-notes)
    - Reviewer may request changes from the contributor to add a release note
    - Reviewer may add a release note by themself in order to unblock the merge process

## Requesting changes

> It's recommended to request changes by submitting `comment` type reviews.

Reviewer: Should we clarify that it doesn't mean that `Request changes` type of reviews should not be submitted. If someone thinks that some thing must be changed before the PR is merged, they should not hesitate adding `Request changes` type reviews.

Author: The issue with that is that person becomes a bottleneck for merging as mentioned below. That's why we should discourage it.

`Request changes` type review would block merging until requester approves the
changes, this can slow down the process if there are multiple reviewers.

## Approving and merging

We use `kueue` bot to merge approved PRs.

If PR is approved, all checks are passing and it has the `kueue` label, it will
be automatically squashed and merged.

> For PRs from Kanister developers, the committer should add the `kueue` label after

Reviewer: ```suggestion
For PRs from Kanister developers, the author should add the `kueue` label after
```

PR was approved.

> For PRs from community, the reviewer should add the `kueue` label.

Reviewer: ```suggestion
For PRs from community members, the reviewer should add the `kueue` label.
```

## Release notes

Kanister is using the [reno](https://docs.openstack.org/reno/latest/) tool to generate changelogs from release note files.

To add release note one could run:

```
make reno-new note=<note_name>
```

Note name should be a short description of a change.

File format is described in [reno docs](https://docs.openstack.org/reno/latest/user/usage.html#editing-a-release-note)

Typical examples would be:

```
---
features:
  - Added new functionality doing X

> ```

Reviewer: is punctuation required in the release notes?

Author: We currently don't check grammar or punctuation. Do we need to? Can you give an example?

Author: There is no requirement for grammar or punctuation in content at the moment.

Reviewer: cool, sounds good.
